### PR TITLE
Calls documentation updates - 2024-05-24

### PR DESCRIPTION
**Pull request by renandincer**

### CODE REVIEW

**Review by kodster28 — CHANGES_REQUESTED** [review]
Heyo @renandincer, generally we don't include openapi specs in our docs.

Is there a reason that these aren't just references out to https://developers.cloudflare.com/api/operations/calls-apps-list and the other API endpoints?

File: content/calls/turn/custom-domains.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+pcx_content_type: get-started
+title: Custom TURN domains
+weight: 15
+---
+
+#  Custom TURN domains
+
+Cloudflare Calls TURN service supports using custom domains. Custom domains do not affect any of of the performance characteristics of Cloudflare Calls TURN.
```
[the remark this answers is not in the document]
Comment (third774) [reply content/calls/turn/custom-domains.md]: Ah, very cool bot! 🙏🏻

**Comment by renandincer** [comment]
@kodster28 good question! This is not a Cloudflare API at api.cloudflare.com. This is a product specific API with a different endpoint. They are not related.

**Comment by kodster28** [comment]
> @kodster28 good question! This is not a Cloudflare API at api.cloudflare.com. This is a product specific API with a different endpoint. They are not related.

Oh... interesting. Gotcha, let me unblock then.

Assuming y'all have a process for updating this when the API changes?

File: content/calls/datachannels.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+pcx_content_type: get-started
+title: Datachannels
+weight: 5
+---
+
+# Datachannels
```
Comment (kodster28) [inline content/calls/datachannels.md]: Assuming this page is still in progress?

File: content/calls/turn/custom-domains.md (hunk 1 above)
Comment (kodster28) [inline content/calls/turn/custom-domains.md]:
```suggestion
Cloudflare Calls TURN service supports using custom domains. Custom domains do not affect any of the performance characteristics of Cloudflare Calls TURN.
```

File: content/calls/turn/custom-domains.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+| STUN over UDP | ✅ | 3478/udp     | 53/udp         |
+| TURN over UDP | ✅ | 3478/udp     | 53 udp         |
+| TURN over TCP | ✅ | 3478/tcp     | 80/tcp         |
+| TURN over TLS | No | 5349/tcp     | 443/tcp        |
+
+## Create a CNAME
+
+To use custom domains for TURN, you must create a CNAME DNS record pointing to `turn.cloudflare.com`. Do not resolve the address of `turn.cloudflare.com` or `stun.cloudflare.com` or use an IP address as the value you input to your DNS record. Only CNAME records are supported.
+
+Any DNS provider, including Cloudflare DNS can be used to set up a CNAME for custom domains. If Cloudflare's DNS is used, the record must be set to ["grey cloud" mode](https://developers.cloudflare.com/dns/manage-dns-records/reference/proxied-dns-records/#dns-only-records).
```
Comment (kodster28) [inline content/calls/turn/custom-domains.md]:
```suggestion
Any DNS provider, including Cloudflare DNS can be used to set up a CNAME for custom domains. If Cloudflare's DNS is used, the record must be [DNS-only](/dns/manage-dns-records/reference/proxied-dns-records/#dns-only-records).
```

File: content/calls/turn/faq.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
+---
+pcx_content_type: get-started
+title: FAQ
+weight: 20
+---
+
+# Frequently Asked Questions
+
+### What is Cloudflare Calls TURN pricing? How exactly is it calculated?
```
Comment (kodster28) [inline content/calls/turn/faq.md:11]: skipped header level

File: content/calls/turn/for-turn-providers.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+---
+pcx_content_type: get-started
+title: For TURN providers
+weight: 18
+---
+
+# Selling Cloudflare Calls TURN
+
+If you are a existing TURN provider but would like to switch to providing Cloudflare Calls TURN for your customers, there is a few considerations.
```
Comment (kodster28) [inline content/calls/turn/for-turn-providers.md]: Feels like there should be more here?

File: content/calls/turn/faq.md (hunk 4 above)
Comment (renandincer) [reply content/calls/turn/faq.md:11]: Yep it's intensional, it's a FAQ and a in between header did not make sense to me. Do you have other styling suggestion for FAQ style?

**Comment by renandincer** [comment]
> > @kodster28 good question! This is not a Cloudflare API at api.cloudflare.com. This is a product specific API with a different endpoint. They are not related.
> 
> Oh... interesting. Gotcha, let me unblock then.
> 
> Assuming y'all have a process for updating this when the API changes?

Yes, this _is_ the source of truth now.

**Comment by renandincer** [comment]
@kodster28 I set `pcx_content_type: get-started` in the markdown header each time, but don't know what it does. Do you have any insight into what this is for?

**Comment by kodster28** [comment]
> @kodster28 I set `pcx_content_type: get-started` in the markdown header each time, but don't know what it does. Do you have any insight into what this is for?

Metadata that we add to the rendered page. It helps us group by the type of content for reporting, etc.
Full list of options + values at https://developers.cloudflare.com/style-guide/documentation-content-strategy/content-types/

File: content/calls/turn/faq.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+Pricing for Cloudflare Calls Serverless SFU and TURN services is $0.05 per GB of data used.
+
+There is a free tier of 1,000 GB before any charges start. This free tier includes both Serverless SFU and TURN services. Cloudflare Calls billing appears as a single line item on your Cloudflare bill, covering both SFU and TURN.
+
+Data traffic between Cloudflare Calls TURN, Cloudflare Calls SFU, and Cloudflare Stream WHIP/WHEP does not incur any charges.
+
+### Does Cloudflare Calls TURN support the expired IETF RFC draft "draft-uberti-behave-turn-rest-00"?
+
+The Cloudflare Calls credential generation function returns a JSON structure similar to the [expired RFC draft "draft-uberti-behave-turn-rest-00"](https://datatracker.ietf.org/doc/html/draft-uberti-behave-turn-rest-00), but it does not include the TTL value. If you need a response in this format, you can modify the JSON from the Cloudflare Calls credential generation endpoint to the required format in your backend server or Cloudflare Workers.
```
Comment (kodster28) [inline content/calls/turn/faq.md:41]: I'd suggest reframing each of these to clearly answer the question in the first sentence.

i.e., if it's `Does TURN support...`, should start with a `Yes` or `No`. It's present in some answers but not in others.

File: content/calls/turn/generate-credentials.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+With a TURN key you can:
+- Generate TURN credentials that expire
+- Revoke previously issued TURN credentials
+
+## Create credentials
+
+You should generate short-lived credentials for each TURN user. In order to create credentials, you should have a back-end service that uses your TURN Token ID and API token to generate credentials. It will make an API call like this:
+
+```sh
```
Comment (kodster28) [inline content/calls/turn/generate-credentials.md:25]:
```suggestion
```bash
```

File: content/calls/turn/generate-credentials.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+The `ttl` value can be adjusted to expire the short lived key in a certain amount of time. This value should be larger than the time you'd expect the users to use the TURN service. For example, if you're using TURN for a video conferencing app, the value should be set to the longest video call you'd expect to happen in the app.
+
+When using short-lived TURN credentials with WebRTC, credentials can be refreshed during a WebRTC session using the `RTCPeerConnection` [`setConfiguration()`](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/setConfiguration) API.
+
+
+## Revoke credentials
+
+Short lived credentials can also be revoked before their TTL expires with a API call like this:
+
+```sh
```
Comment (kodster28) [inline content/calls/turn/generate-credentials.md:85]:
```suggestion
```bash
```

File: content/calls/turn/faq.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+weight: 20
+---
+
+# Frequently Asked Questions
+
+### What is Cloudflare Calls TURN pricing? How exactly is it calculated?
+
+Cloudflare TURN pricing is based on the data sent from the Cloudflare edge to the TURN client, as described in [RFC 8656 Figure 1](https://datatracker.ietf.org/doc/html/rfc8656#fig-turn-model). This means data sent from the TURN server to the TURN client.
+
+Pricing for Cloudflare Calls Serverless SFU and TURN services is $0.05 per GB of data used.
```
Comment (kodster28) [inline content/calls/turn/faq.md:15]: You may want to standardize the pricing / limits info on a separate page.

Sorta stuff that gets very tangled / hard to update if it doesn't live in a centralized place (saw it in a few other locations).

**Review by kodster28 — APPROVED** [review]
Added a few other comments / suggestions

File: content/calls/turn/faq.md (hunk 9 above)
Comment (renandincer) [reply content/calls/turn/faq.md:15]: Makes sense.

**Pull request merged**